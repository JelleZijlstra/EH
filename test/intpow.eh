#!/usr/bin/ehi

intpow(m, n) = if n == 0
	1
elsif n == 1
	m
else
	part = intpow(m, n / 2)
	if n % 2 == 1
		part * part * m
	else
		part * part
	end
end

echo(intpow(2, 16))
echo(intpow(3, 7))
