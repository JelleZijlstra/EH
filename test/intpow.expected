%%stdout%%
65536
2187
%%stderr%%
